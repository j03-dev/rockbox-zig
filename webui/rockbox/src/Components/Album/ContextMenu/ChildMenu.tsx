import { useTheme } from "@emotion/react";
import { StatefulMenu } from "baseui/menu";
import { FC } from "react";

const ChildMenu: FC<{
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  recentPlaylists: any[];
  onSelect: (item: { id: string; label: string }) => void;
}> = ({ onSelect, recentPlaylists }) => {
  const theme = useTheme();
  const items =
    recentPlaylists.length > 0
      ? {
          __ungrouped: [
            {
              label: "Create new playlist",
            },
          ],
          RECENT: recentPlaylists.map((playlist) => ({
            id: playlist.id,
            label: <div>{playlist.name}</div>,
          })),
        }
      : {
          __ungrouped: [
            {
              label: "Create new playlist",
            },
          ],
        };
  return (
    <StatefulMenu
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      items={items as any}
      overrides={{
        List: {
          style: {
            boxShadow: "none",
            backgroundColor: theme.colors.popoverBackground,
          },
        },
      }}
      onItemSelect={({ item }) => onSelect(item)}
    />
  );
};

export default ChildMenu;
