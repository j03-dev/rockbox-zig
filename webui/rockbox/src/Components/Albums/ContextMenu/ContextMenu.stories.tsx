import type { Meta, StoryObj } from "@storybook/react";

import ContextMenu from "./ContextMenu";
import { fn } from "@storybook/test";

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
const meta = {
  title: "Components/Albums/ContextMenu",
  component: ContextMenu,
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ["autodocs"],
  // More on argTypes: https://storybook.js.org/docs/api/argtypes
} satisfies Meta<typeof ContextMenu>;

export default meta;
type Story = StoryObj<typeof meta>;

// More on writing stories with args: https://storybook.js.org/docs/writing-stories/args
export const Default: Story = {
  args: {
    album: {
      title: "Pluto x Baby Pluto",
      artist: "Future, Lil Uzi Vert",
      cover:
        "https://resources.tidal.com/images/fe6787d5/4ba5/4d3e/8576/48943ee6a768/320x320.jpg",
    },
    onPlayNext: fn(),
    onCreatePlaylist: fn(),
    onAddTrackToPlaylist: fn(),
    onPlayLast: fn(),
    onAddShuffled: fn(),
    onPlayLastShuffled: fn(),
    onPlayShuffled: fn(),
    recentPlaylists: [],
  },
};
