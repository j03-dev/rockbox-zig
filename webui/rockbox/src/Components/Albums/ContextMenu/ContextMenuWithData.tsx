import { FC } from "react";
import ContextMenu from "./ContextMenu";
import {
  useInsertAlbumMutation,
  usePlayAlbumMutation,
} from "../../../Hooks/GraphQL";
import {
  PLAYLIST_INSERT_FIRST,
  PLAYLIST_INSERT_LAST,
  PLAYLIST_INSERT_LAST_SHUFFLED,
  PLAYLIST_INSERT_SHUFFLED,
} from "../../../Types/playlist";

export type ContextMenuWithDataProps = {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  item: any;
};

const ContextMenuWithData: FC<ContextMenuWithDataProps> = ({ item }) => {
  const [playAlbum] = usePlayAlbumMutation();
  const [insertAlbum] = useInsertAlbumMutation();

  const onPlayShuffled = (albumId: string) => {
    playAlbum({
      variables: {
        albumId,
        shuffle: true,
      },
    });
  };

  const onPlayNext = (albumId: string) => {
    insertAlbum({
      variables: {
        albumId,
        position: PLAYLIST_INSERT_FIRST,
      },
    });
  };

  const onPlayLast = (albumId: string) => {
    insertAlbum({
      variables: {
        albumId,
        position: PLAYLIST_INSERT_LAST,
      },
    });
  };

  const onAddShuffled = (albumId: string) => {
    insertAlbum({
      variables: {
        albumId,
        position: PLAYLIST_INSERT_SHUFFLED,
      },
    });
  };

  const onPlayLastShuffled = (albumId: string) => {
    insertAlbum({
      variables: {
        albumId,
        position: PLAYLIST_INSERT_LAST_SHUFFLED,
      },
    });
  };

  return (
    <ContextMenu
      album={item}
      onPlayNext={onPlayNext}
      onCreatePlaylist={() => {}}
      onAddTrackToPlaylist={() => {}}
      onPlayLast={onPlayLast}
      onAddShuffled={onAddShuffled}
      onPlayLastShuffled={onPlayLastShuffled}
      onPlayShuffled={onPlayShuffled}
      recentPlaylists={[]}
    />
  );
};

export default ContextMenuWithData;
