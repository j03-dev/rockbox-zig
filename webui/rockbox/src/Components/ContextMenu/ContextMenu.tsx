/* eslint-disable @typescript-eslint/no-explicit-any */
import Add from "../Icons/Add";
import Heart from "../Icons/Heart";
import HeartOutline from "../Icons/HeartOutline";
import { EllipsisHorizontal } from "@styled-icons/ionicons-sharp";
import { StatefulPopover } from "baseui/popover";
import { NestedMenus, StatefulMenu } from "baseui/menu";
import TrackIcon from "../Icons/Track";
import { useTheme } from "@emotion/react";
import ChildMenu from "./ChildMenu";
import { FC, useState } from "react";
import {
  AlbumCover,
  AlbumCoverAlt,
  Artist,
  Container,
  Hover,
  Icon,
  Separator,
  Title,
  Track,
  TrackInfos,
} from "./styles";

export type ContextMenuProps = {
  liked?: boolean;
  track: any;
  onPlayNext: (id: string) => void;
  onCreatePlaylist: (name: string, description?: string) => void;
  onAddTrackToPlaylist: (playlistId: string, trackId: string) => void;
  onPlayLast: (id: string) => void;
  onAddShuffled: (id: string) => void;
  recentPlaylists: any[];
};

const ContextMenu: FC<ContextMenuProps> = ({
  liked = false,
  track,
  onPlayNext,
  // onCreatePlaylist,
  onPlayLast,
  onAddTrackToPlaylist,
  onAddShuffled,
  recentPlaylists,
}) => {
  const theme = useTheme();
  const [, setIsNewPlaylistModalOpen] = useState(false);
  return (
    <Container>
      <Hover>
        <StatefulPopover
          placement="left"
          autoFocus={false}
          content={({ close }) => (
            <div
              style={{
                width: 205,
              }}
            >
              <Track>
                {track.cover && <AlbumCover src={track.cover} />}
                {!track.cover && (
                  <AlbumCoverAlt>
                    <TrackIcon width={24} height={24} color="#a4a3a3" />
                  </AlbumCoverAlt>
                )}
                <TrackInfos>
                  <Title>{track.title}</Title>
                  <Artist>{track.artist}</Artist>
                </TrackInfos>
              </Track>
              <NestedMenus>
                <StatefulMenu
                  overrides={{
                    List: {
                      style: {
                        boxShadow: "none",
                        backgroundColor: theme.colors.popoverBackground,
                      },
                    },
                    Option: {
                      props: {
                        getChildMenu: (item: { label: string }) => {
                          if (item.label === "Add to Playlist") {
                            return (
                              <ChildMenu
                                recentPlaylists={recentPlaylists}
                                onSelect={(item: {
                                  id: string;
                                  label: string;
                                }) => {
                                  if (item.label === "Create new playlist") {
                                    setIsNewPlaylistModalOpen(true);
                                  } else {
                                    onAddTrackToPlaylist(item.id, track.id);
                                  }
                                  close();
                                }}
                              />
                            );
                          }
                          return null;
                        },
                      },
                    },
                  }}
                  items={[
                    {
                      id: "1",
                      label: "Play Next",
                    },
                    {
                      id: "2",
                      label: "Add to Playlist",
                    },
                    {
                      id: "3",
                      label: "Play Last",
                    },
                    {
                      id: "4",
                      label: "Add Shuffled",
                    },
                  ]}
                  onItemSelect={({ item }) => {
                    if (item.label === "Add to Playlist") {
                      return;
                    }
                    if (item.label === "Play Next") {
                      onPlayNext(track.id);
                    }
                    if (item.label === "Play Last") {
                      onPlayLast(track.id);
                    }
                    if (item.label === "Add Shuffled") {
                      onAddShuffled(track.id);
                    }
                    close();
                  }}
                />
              </NestedMenus>
            </div>
          )}
          overrides={{
            Inner: {
              style: {
                backgroundColor: theme.colors.popoverBackground,
              },
            },
          }}
        >
          <Icon>
            <EllipsisHorizontal />
          </Icon>
        </StatefulPopover>
      </Hover>
      <Separator />
      <StatefulPopover
        autoFocus={false}
        placement="left"
        content={({ close }) => (
          <div style={{ width: 205 }}>
            <StatefulMenu
              overrides={{
                List: {
                  style: {
                    boxShadow: "none",
                    backgroundColor: theme.colors.popoverBackground,
                  },
                },
              }}
              items={[
                {
                  id: "1",
                  label: <div>Create new playlist</div>,
                },
              ]}
              onItemSelect={() => {
                setIsNewPlaylistModalOpen(true);
                close();
              }}
            />
          </div>
        )}
        overrides={{
          Inner: {
            style: {
              backgroundColor: theme.colors.popoverBackground,
            },
          },
        }}
      >
        <Icon>
          <Add size={24} color={theme.colors.icon} />
        </Icon>
      </StatefulPopover>
      <Separator />
      {liked && (
        <Icon>
          <Heart height={24} width={24} color={theme.colors.icon} />
        </Icon>
      )}
      {!liked && (
        <Icon>
          <HeartOutline height={24} width={24} color={theme.colors.icon} />
        </Icon>
      )}
      {/*<NewPlaylistModal
        onClose={() => {
          setIsNewPlaylistModalOpen(false);
        }}
        isOpen={isNewPlaylistModalOpen}
        onCreatePlaylist={onCreatePlaylist}
      />
      */}
    </Container>
  );
};

export default ContextMenu;
