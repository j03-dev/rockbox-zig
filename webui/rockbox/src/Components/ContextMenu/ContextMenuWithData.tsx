import { FC } from "react";
import ContextMenu from "./ContextMenu";
import {
  useGetLikedTracksQuery,
  useInsertTracksMutation,
  useLikeTrackMutation,
  useUnlikeTrackMutation,
} from "../../Hooks/GraphQL";
import {
  PLAYLIST_INSERT_FIRST,
  PLAYLIST_INSERT_LAST,
  PLAYLIST_INSERT_SHUFFLED,
} from "../../Types/playlist";
import { useRecoilState } from "recoil";
import { likesState } from "../Likes/LikesState";

export type ContextMenuWithDataProps = {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  track: any;
};

const ContextMenuWithData: FC<ContextMenuWithDataProps> = ({ track }) => {
  const [likes, setLikes] = useRecoilState(likesState);
  const { refetch } = useGetLikedTracksQuery({
    fetchPolicy: "network-only",
  });
  const [insertTracks] = useInsertTracksMutation();
  const [likeTrack] = useLikeTrackMutation();
  const [unlikeTrack] = useUnlikeTrackMutation();

  const onPlayNext = (path: string) => {
    insertTracks({
      variables: {
        position: PLAYLIST_INSERT_FIRST,
        tracks: [path],
      },
    });
  };

  const onPlayLast = (path: string) => {
    insertTracks({
      variables: {
        position: PLAYLIST_INSERT_LAST,
        tracks: [path],
      },
    });
  };

  const onAddShuffled = (path: string) => {
    insertTracks({
      variables: {
        position: PLAYLIST_INSERT_SHUFFLED,
        tracks: [path],
      },
    });
  };

  const onLike = async (trackId: string) => {
    setLikes({
      ...likes,
      [trackId]: true,
    });
    await likeTrack({
      variables: {
        trackId,
      },
    });
    await refetch();
  };

  const onUnlike = async (trackId: string) => {
    setLikes({
      ...likes,
      [trackId]: false,
    });
    await unlikeTrack({
      variables: {
        trackId,
      },
    });
    await refetch();
  };

  return (
    <ContextMenu
      track={track}
      onPlayNext={onPlayNext}
      onCreatePlaylist={() => {}}
      onAddTrackToPlaylist={() => {}}
      onPlayLast={onPlayLast}
      onAddShuffled={onAddShuffled}
      onLike={onLike}
      onUnlike={onUnlike}
      recentPlaylists={[]}
      liked={likes[track.id]}
    />
  );
};

export default ContextMenuWithData;
