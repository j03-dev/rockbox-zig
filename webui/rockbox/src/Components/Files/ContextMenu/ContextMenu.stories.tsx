import type { Meta, StoryObj } from "@storybook/react";

import ContextMenu from "./ContextMenu";
import { fn } from "@storybook/test";

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
const meta = {
  title: "Components/Files/ContextMenu",
  component: ContextMenu,
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ["autodocs"],
  // More on argTypes: https://storybook.js.org/docs/api/argtypes
} satisfies Meta<typeof ContextMenu>;

export default meta;
type Story = StoryObj<typeof meta>;

// More on writing stories with args: https://storybook.js.org/docs/writing-stories/args
export const Directory: Story = {
  args: {
    entry: {
      title: "[E]  Live From The Underground [15421743] [2012]",
      isDirectory: true,
    },
    onPlayNext: fn(),
    onCreatePlaylist: fn(),
    onAddTrackToPlaylist: fn(),
    onPlayLast: fn(),
    onAddShuffled: fn(),
    onPlayShuffled: fn(),
    onPlayLastShuffled: fn(),
    recentPlaylists: [],
  },
};
