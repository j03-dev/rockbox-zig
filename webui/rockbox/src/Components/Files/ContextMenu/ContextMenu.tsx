/* eslint-disable @typescript-eslint/no-explicit-any */
import { EllipsisHorizontal } from "@styled-icons/ionicons-sharp";
import { Folder2 } from "@styled-icons/bootstrap";
import { StatefulPopover } from "baseui/popover";
import { NestedMenus, StatefulMenu } from "baseui/menu";
import TrackIcon from "../../Icons/Track";
import { useTheme } from "@emotion/react";
import ChildMenu from "./ChildMenu";
import { FC, useMemo, useState } from "react";
import {
  AlbumCoverAlt,
  Artist,
  Container,
  Hover,
  Icon,
  Title,
  Track,
  TrackInfos,
} from "./styles";

export type ContextMenuProps = {
  entry: any;
  onPlayNext: (path: string) => void;
  onCreatePlaylist: (name: string, description?: string) => void;
  onAddTrackToPlaylist: (playlistId: string, path: string) => void;
  onPlayLast: (path: string) => void;
  onAddShuffled: (path: string) => void;
  onPlayShuffled: (path: string) => void;
  onPlayLastShuffled: (path: string) => void;
  recentPlaylists: any[];
};

const ContextMenu: FC<ContextMenuProps> = ({
  entry,
  onPlayNext,
  // onCreatePlaylist,
  onPlayLast,
  onAddTrackToPlaylist,
  onAddShuffled,
  onPlayShuffled,
  onPlayLastShuffled,
  recentPlaylists,
}) => {
  const theme = useTheme();
  const [, setIsNewPlaylistModalOpen] = useState(false);
  const items = useMemo(() => {
    if (entry.isDirectory) {
      return [
        {
          id: "1",
          label: "Play Next",
        },
        {
          id: "2",
          label: "Add to Playlist",
        },
        {
          id: "3",
          label: "Play Last",
        },
        {
          id: "4",
          label: "Add Shuffled",
        },
        {
          id: "5",
          label: "Play Last Shuffled",
        },
        {
          id: "6",
          label: "Play Shuffled",
        },
      ];
    }
    return [
      {
        id: "1",
        label: "Play Next",
      },
      {
        id: "2",
        label: "Add to Playlist",
      },
      {
        id: "3",
        label: "Play Last",
      },
      {
        id: "4",
        label: "Add Shuffled",
      },
    ];
  }, [entry]);
  return (
    <Container>
      <Hover>
        <StatefulPopover
          placement="left"
          autoFocus={false}
          content={({ close }) => (
            <div
              style={{
                width: 205,
              }}
            >
              <Track>
                {entry.isDirectory && (
                  <AlbumCoverAlt>
                    <Folder2 size={18} />
                  </AlbumCoverAlt>
                )}
                {!entry.isDirectory && (
                  <AlbumCoverAlt>
                    <TrackIcon width={24} height={24} color="#a4a3a3" />
                  </AlbumCoverAlt>
                )}
                <TrackInfos>
                  <Title>{entry.title}</Title>
                  <Artist>{entry.artist}</Artist>
                </TrackInfos>
              </Track>
              <NestedMenus>
                <StatefulMenu
                  overrides={{
                    List: {
                      style: {
                        boxShadow: "none",
                        backgroundColor: theme.colors.popoverBackground,
                      },
                    },
                    Option: {
                      props: {
                        getChildMenu: (item: { label: string }) => {
                          if (item.label === "Add to Playlist") {
                            return (
                              <ChildMenu
                                recentPlaylists={recentPlaylists}
                                onSelect={(item: {
                                  id: string;
                                  label: string;
                                }) => {
                                  if (item.label === "Create new playlist") {
                                    setIsNewPlaylistModalOpen(true);
                                  } else {
                                    onAddTrackToPlaylist(item.id, entry.id);
                                  }
                                  close();
                                }}
                              />
                            );
                          }
                          return null;
                        },
                      },
                    },
                  }}
                  items={items}
                  onItemSelect={({ item }) => {
                    switch (item.label) {
                      case "Play Next":
                        onPlayNext(entry.path);
                        break;
                      case "Play Last":
                        onPlayLast(entry.path);
                        break;
                      case "Add Shuffled":
                        onAddShuffled(entry.path);
                        break;
                      case "Play Shuffled":
                        onPlayShuffled(entry.path);
                        break;
                      case "Play Last Shuffled":
                        onPlayLastShuffled(entry.path);
                        break;
                      default:
                        break;
                    }
                    close();
                  }}
                />
              </NestedMenus>
            </div>
          )}
          overrides={{
            Inner: {
              style: {
                backgroundColor: theme.colors.popoverBackground,
              },
            },
            Body: {
              style: {
                zIndex: 1,
              },
            },
          }}
        >
          <Icon>
            <EllipsisHorizontal size={24} />
          </Icon>
        </StatefulPopover>
      </Hover>
      {/*<NewPlaylistModal
        onClose={() => {
          setIsNewPlaylistModalOpen(false);
        }}
        isOpen={isNewPlaylistModalOpen}
        onCreatePlaylist={onCreatePlaylist}
      />
      */}
    </Container>
  );
};

export default ContextMenu;
