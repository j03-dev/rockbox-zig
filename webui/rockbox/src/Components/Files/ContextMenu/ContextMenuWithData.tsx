import { FC } from "react";
import ContextMenu from "./ContextMenu";
import {
  useInsertDirectoryMutation,
  useInsertTracksMutation,
  usePlayDirectoryMutation,
} from "../../../Hooks/GraphQL";
import {
  PLAYLIST_INSERT_FIRST,
  PLAYLIST_INSERT_LAST,
  PLAYLIST_INSERT_LAST_SHUFFLED,
  PLAYLIST_INSERT_SHUFFLED,
} from "../../../Types/playlist";

export type ContextMenuWithDataProps = {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  entry: any;
};

const ContextMenuWithData: FC<ContextMenuWithDataProps> = ({ entry }) => {
  const [insertTracks] = useInsertTracksMutation();
  const [playDirectory] = usePlayDirectoryMutation();
  const [insertDirectory] = useInsertDirectoryMutation();

  const onPlayNext = (path: string) => {
    if (!entry.isDirectory) {
      insertTracks({
        variables: {
          tracks: [path],
          position: PLAYLIST_INSERT_FIRST,
        },
      });
      return;
    }
    insertDirectory({
      variables: {
        position: PLAYLIST_INSERT_FIRST,
        directory: path,
      },
    });
  };

  const onPlayLast = (path: string) => {
    if (!entry.isDirectory) {
      insertTracks({
        variables: {
          tracks: [path],
          position: PLAYLIST_INSERT_LAST,
        },
      });
      return;
    }
    insertDirectory({
      variables: {
        position: PLAYLIST_INSERT_LAST,
        directory: path,
      },
    });
  };

  const onAddShuffled = (path: string) => {
    if (!entry.isDirectory) {
      insertTracks({
        variables: {
          tracks: [path],
          position: PLAYLIST_INSERT_SHUFFLED,
        },
      });
      return;
    }
    insertDirectory({
      variables: {
        position: PLAYLIST_INSERT_SHUFFLED,
        directory: path,
      },
    });
  };

  const onPlayShuffled = (path: string) => {
    playDirectory({
      variables: {
        path,
        shuffle: true,
        recurse: true,
      },
    });
  };

  const onPlayLastShuffled = (path: string) => {
    insertDirectory({
      variables: {
        position: PLAYLIST_INSERT_LAST_SHUFFLED,
        directory: path,
      },
    });
  };

  return (
    <ContextMenu
      entry={entry}
      onPlayNext={onPlayNext}
      onCreatePlaylist={() => {}}
      onAddTrackToPlaylist={() => {}}
      onPlayLast={onPlayLast}
      onAddShuffled={onAddShuffled}
      onPlayShuffled={onPlayShuffled}
      onPlayLastShuffled={onPlayLastShuffled}
      recentPlaylists={[]}
    />
  );
};

export default ContextMenuWithData;
