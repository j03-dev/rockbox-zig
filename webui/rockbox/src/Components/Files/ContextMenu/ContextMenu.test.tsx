import { render } from "@testing-library/react";
import ContextMenu from "./ContextMenu";
import Providers from "../../../Providers";
import { vi } from "vitest";

describe("ContextMenu", () => {
  it("should render", () => {
    const { container } = render(
      <Providers>
        <ContextMenu
          entry={{
            title: "[E]  Live From The Underground [15421743] [2012]",
            isDirectory: true,
          }}
          onPlayNext={vi.fn()}
          onCreatePlaylist={vi.fn()}
          onAddTrackToPlaylist={vi.fn()}
          onAddShuffled={vi.fn()}
          onPlayShuffled={vi.fn()}
          onPlayLast={vi.fn()}
          onPlayLastShuffled={vi.fn()}
          recentPlaylists={[]}
        />
      </Providers>
    );
    expect(container).toMatchSnapshot();
  });
});
