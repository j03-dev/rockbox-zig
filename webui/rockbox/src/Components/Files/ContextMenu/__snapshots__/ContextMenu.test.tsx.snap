// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContextMenu > should render 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="css-113iyi7"
    >
      <button
        class="css-1t1di7c"
      >
        <div
          aria-expanded="false"
          aria-haspopup="true"
          class="css-l6q2r8"
        >
          <svg
            aria-hidden="true"
            class="StyledIconBase-sc-ea9ulj-0 dmvaRK"
            fill="currentColor"
            focusable="false"
            height="24"
            viewBox="0 0 512 512"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="256"
              cy="256"
              r="48"
            />
            <circle
              cx="416"
              cy="256"
              r="48"
            />
            <circle
              cx="96"
              cy="256"
              r="48"
            />
          </svg>
        </div>
      </button>
    </div>
  </div>
  <div
    class=""
  />
</div>
`;
